src/amp/lib/permutive.ts

```diff
@@ -0,0 +1,39 @@
+export interface PermutivePayload {
+    'properties.content.Premium'?: string;
+    'properties.content.id'?: string;
+    'properties.content.title'?: string;
+    'properties.content.section'?: string;
+    'properties.content.authors!'?: string;
+    'properties.content.keywords!'?: string;
+    'properties.content.publishedAt'?: string;
+    'properties.user.edition'?: string;
+}
+
+export const generatePermutivePayload = (
+    rawConfig: CommercialConfigType,
+): PermutivePayload => {
+    const publishedAt =
+        rawConfig.webPublicationDate &&
+        typeof rawConfig.webPublicationDate === 'number'
+            ? new Date(rawConfig.webPublicationDate).toISOString()
+            : void 0;
+    const config: { [key: string]: any } = {
+        'properties.content.Premium': rawConfig.isPaidContent,
+        'properties.content.id': rawConfig.pageId,
+        'properties.content.title': rawConfig.headline,
+        'properties.content.section': rawConfig.section,
+        'properties.content.authors!list[string]': rawConfig.author,
+        'properties.content.keywords!list[string]': rawConfig.keywords,
+        'properties.content.publishedAt': publishedAt,
+        'properties.user.edition': rawConfig.edition,
```

Comment:
We may need to make sure the format of the edition is correct as well. It needs to be precisely one of:
* "UK"
* "US"
* "AU"
* "INT"

Reply:
DCR has the type `Edition` that would achieve this

Reply:
@adamnfish when you same make sure, do you mean assert it's all caps or something else?
@oliverlloyd I can see this type used in CAPIType but not in the ConfigType. Is Config derived from CAPI anyway?
We don't validate `edition` in `frontend` so we need to change things there too.